packages/@lwc/template-compiler/src/codegen/helpers.ts

```diff
@@ -70,25 +87,39 @@ export function containsDynamicChildren(children: ChildNode[]): boolean {
 
         return false;
     });
+
+    // In order to check the if-elseif-else chain fully, if the parent is an ElseIfBlock
+    // the else branch must be checked as well.
+    // Note that the ElseIfBlock can only occur when it is inside an if-elseif-else chain
+    // meaning that the children passed from transformChildren will not contain an ElseifBlock or ElseBlock
+    const elseIfHasDynamicChildren =
+        isElseifBlock(parent) && parent.else ? containsDynamicChildren(parent.else) : false;
```

Comment:
Isn't this case also needed if `isIfBlock(parent)`?
If `typeof parent === IfBlock`, where are we checking the `else` branch of that node?

Reply:
Good call, I've updated the code to look for the `IfBlock` as well.